microsoft-edge/progressive-web-apps-chromium/how-to/widgets.md

```diff
@@ -0,0 +1,389 @@
+| `icons` | An array of icons to be used for the Widget. If not present, the manifest `icons` member will be used. | No |
+| `backgrounds` | An array of background images that can be used in the template (if the template supports it). | No |
+
+
+<!-- ====================================================================== -->
+## Choose or define a Widget template
+
+**OPEN QUESTIONS: Are generic templates supported at all? Should they be documented?**
+
+To make Widgets easy to create and adapt to various operating systems, they are displayed by using templates. Two types of templates exist:
+
+* Generic templates, defined by their names using the `template` field.
+* Custom templates, defined by their URLs using a custom template field.
+
+For the time being, only custom Adaptive Cards templates are supported. Adaptive Cards are an open card exchange format that can be used to exchange UI content in a common and consistent way. See [Adaptive Cards Overview](/adaptive-cards/).
+
+To define a custom Adaptive Cards template, use the `ms_ac_template` field:
+
+```json
+{
+  "name": "Agenda",
+  "description": "Your day, at a glance",
+  "tag": "agenda",
+  "data": "/widgets/data/agenda",
+  "type": "application/json",
+  "ms_ac_template": "/widgets/template/agenda_template.ac.json"
+}
+```
+
+The `ms_ac_template` field value should be a valid URL to a template file.
+
+Here is an example of an Adaptive Cards template:
+
+```json
+{
+  "type": "AdaptiveCard",
+  "body": [
+    {
+      "type": "TextBlock",
+      "size": "Medium",
+      "text": "Hello ${name}"
+    },
+    {
+      "type": "TextBlock",
+      "text": "You have ${meetingNb} meetings today"
+    }
+  ],
+  "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
+  "version": "1.5"
+}
+```
+
+To learn more, see [Adaptive Cards Templating](/adaptive-cards/templating/).
+
+
+<!-- ====================================================================== -->
+## Bind data to your template
+
+The template declares the user interface of a Widget.  Data then populates this user interface.
+
+To bind data to your template, use the `data` field in your Widget definition. This field should be set to a URL that returns valid JSON data.
+
+The template defined in [the previous section](#choose-or-define-a-widget-template) contains two variables: `name` and `meetingNb`, which are enclosed in the binding expression syntax: `${}`. The data that's returned by the `data` URL in your Widget definition should contain values for these variables.
+
+Here is an example of what the `data` URL might return: 
+
+```json
+{
+  "name": "Amelie Garner",
+  "meetingNb": 3
+}
+```
+
+
+<!-- ====================================================================== -->
+## Define Widget actions
+
+**OPEN QUESTIONS: Are actions supported yet? How do actions from the manifest work with actions from the ac template? Maybe using the data.tag field in the ac action?**
```

Comment:
Yes, custom `WidgetActions` will be supported. The `WidgetAction.action` maps to an action's `verb`:

```
 "actions": [
        {
          "type": "Action.Execute",
          "title": "Submit",
          "verb": "personalDetailsFormSubmit",
          "fallback": "Action.Submit"
        }
      ]
```